src/coreclr/src/jit/importer.cpp

```diff
@@ -4015,6 +4015,45 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 break;
             }
 
+            case NI_System_Type_IsAssignableFrom:
+            {
+                // Optimize `typeof(TTo).IsAssignableFrom(TTFrom)` to true/false
+                GenTree* typeTo   = impStackTop(1).val;
+                GenTree* typeFrom = impStackTop(0).val;
+
+                if (typeTo->IsCall() && typeFrom->IsCall())
+                {
+                    // make sure both arguments are `typeof()`
```

Comment:
Do we know what typical uses of `IsAssignableFrom` look like? Are they usually  via `typeof` or does `GetType` show up with any frequency?

Reply:
Good point, I started with typeof()+typeof() but it seems it makes sense to handle more cases, will  analyze usages in popular projects.

Reply:
This is the scenario I was thinking about
https://github.com/dotnet/coreclr/issues/2591#issuecomment-170209816

Reply:
Currently this code handles two cases:
```csharp
bool a = typeof(T1).IsAssignableFrom(typeof(T2));

bool a = variable.GetType().IsAssignableFrom(typeof(T2));
```